native/src/components/SearchInput.tsx

```diff
@@ -26,46 +18,36 @@ export const Wrapper = styled.View<{ space: boolean }>`
   padding: 10px 0;
   background-color: ${props => props.theme.colors.backgroundColor};
 `
-export const SearchIcon = styled(Icon).attrs(props => ({
-  name: 'search',
-  size: 30,
-  color: props.theme.colors.textSecondaryColor
-}))``
 
 type PropsType = {
   placeholderText: string
   filterText: string
-  onFilterTextChange: (arg0: string) => void
+  onFilterTextChange: (filterText: string) => void
   spaceSearch: boolean
-  theme: ThemeType
 }
 
-class SearchInput extends React.Component<PropsType> {
-  static defaultProps = {
-    spaceSearch: false
-  }
-
-  onFilterTextChange = (text: string): void => {
-    const { onFilterTextChange } = this.props
-    onFilterTextChange(text)
-  }
-
-  render(): ReactNode {
-    const { filterText, placeholderText, theme, spaceSearch } = this.props
-    return (
-      <Wrapper theme={theme} space={spaceSearch}>
-        <SearchIcon theme={theme} />
-        <Input
-          {...testID('Search-Input')}
-          theme={theme}
-          placeholder={placeholderText}
-          aria-label={placeholderText}
-          defaultValue={filterText}
-          onChangeText={this.onFilterTextChange}
-        />
-      </Wrapper>
-    )
-  }
+const SearchInput = ({
+  placeholderText,
+  filterText,
+  onFilterTextChange,
+  spaceSearch = false
+}: PropsType): ReactElement => {
+  const theme = useTheme()
+
+  return (
+    <Wrapper space={spaceSearch}>
+      <Icon name='search' size={30} color={theme.colors.textSecondaryColor} />
+      <Input
+        {...testID('Search-Input')}
+        multiline={false}
+        placeholderTextColor={theme.colors.textSecondaryColor}
```

Comment:
no need to pass this attribute to a styled component

Reply:
It is not passed to the styled component. This is just a regular prop of a react-native component and not a style prop.